src/sage/geometry/hyperplane_arrangement/arrangement.py

```diff
@@ -3596,6 +3663,395 @@ def derivation_module_basis(self, algorithm="singular"):
+            True
+            sage: T1.isomorphism(M2)                                                    # optional - sage.graphs
+            {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5}
+            sage: a0 = hyperplane_arrangements.semiorder(3); a0                         # optional - sage.combinat
+            Arrangement of 6 hyperplanes of dimension 3 and rank 2
+            sage: L.<t0, t1, t2> = OrderedHyperplaneArrangements(QQ)
+            sage: a = L(a0)                                                             # optional - sage.combinat
+            sage: ca = a.cone()                                                         # optional - sage.combinat
+            sage: m = ca.matroid()                                                      # optional - sage.combinat
+            sage: a1 = a.hyperplane_section(proj=False)                                 # optional - sage.combinat
+            sage: a1                                                                    # optional - sage.combinat
+            Arrangement of 6 hyperplanes of dimension 2 and rank 2
+            sage: ca1 = a1.cone()                                                       # optional - sage.combinat
+            sage: m1 = ca1.matroid()                                                    # optional - sage.combinat
+            sage: m.isomorphism(m1)                                                     # optional - sage.combinat
+            {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6}
+            sage: p0 = hyperplane_arrangements.Shi(4)                                   # optional - sage.combinat
+            sage: L.<t0, t1, t2, t3> = OrderedHyperplaneArrangements(QQ)
+            sage: p = L(p0)                                                             # optional - sage.combinat
+            sage: a = p.hyperplane_section(proj=False); a                               # optional - sage.combinat
+            Arrangement of 12 hyperplanes of dimension 3 and rank 3
+            sage: ca = a.cone()                                                         # optional - sage.combinat
+            sage: m = ca.matroid().truncation()                                         # optional - sage.combinat
+            sage: a1 = a.hyperplane_section(proj=False); a1                             # optional - sage.combinat
+            Arrangement of 12 hyperplanes of dimension 2 and rank 2
+            sage: ca1 = a1.cone()                                                       # optional - sage.combinat
+            sage: m1 = ca1.matroid()                                                    # optional - sage.combinat
+            sage: m1.is_isomorphism(m, {j: j for j in range(13)})                       # optional - sage.combinat
+            True
+        """
+        from sage.matrix.constructor import Matrix
+        if proj and not self.is_central():
+            raise TypeError('The arrangement is not central')
```

Comment:
```suggestion
            raise TypeError('the arrangement is not central')
```
and add a doctest showing this error.

Reply:
Thanks, I need `is_linear` instead `is_central`. Doctest added